src/checkout.c

```diff
@@ -243,6 +243,12 @@ static int checkout_action_common(
 		if (delta->new_file.mode == GIT_FILEMODE_LINK && wd != NULL)
 			*action |= CHECKOUT_ACTION__REMOVE;
 
+		/* if the file is on disk and doesn't match our mode, force update */
+		if (wd &&
+			GIT_PERMS_IS_EXEC(wd->mode) ^
+			GIT_PERMS_IS_EXEC(delta->new_file.mode))
```

Comment:
I'm wondering if this has an adavantage over writing `!=` since we're dealing with a boolean on each side, no?